### ISSUE TITLE
Using a SAS token with private blobs

### ISSUE THREAD

**Issue by roycornelissen** [jakeFeldman/strapi-provider-upload-azure-storage#2]
We're using this provider to store our images in a private blob storage. On our front end site, we use SAS tokens to resolve those images, but in the Strapi CMS Admin backend, this is not the case. All files now give `404 Not Found` because of that. Since the provider has the private access key, can it be made to use that access key to add SAS urls so that the images can be returned?

**Comment by jakeFeldman** [comment]
Hello Roy thanks for reaching out, I'm not too familiar with Azure SAS tokens and if Strapi would support this. I would happily accept a PR to add the functionality you're looking for as long as it continues to function with private access keys as well.

**Comment by roycornelissen** [comment]
Hi Jake, thanks for looking into this. I did some more investigation and I guess there is a piece missing in the puzzle here, which is that Strapi does not have a built in `download` hook where this provider can tap in like with the `upload` and `delete`.

I made a feature request for that on the strapi roadmap, then I’ll be happy to make a PR for that if they decide to add that to Strapi. I think that should be possible with the private access keys.

Thanks for making this provider!
